lib/Sema/CSSimplify.cpp

```diff
@@ -1230,8 +1250,85 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
   auto argumentLocator = locator.withPathElement(
       ConstraintLocator::FunctionArgument);
 
-  if (func1Params.size() != func2Params.size())
-    return getTypeMatchFailure(argumentLocator);
+  int diff = func1Params.size() - func2Params.size();
+  if (diff != 0) {
+    if (!shouldAttemptFixes())
+      return getTypeMatchFailure(argumentLocator);
+
+    auto *anchor = locator.trySimplifyToExpr();
+    if (!anchor)
+      return getTypeMatchFailure(argumentLocator);
+
+    // If there are missing arguments, let's add them
+    // using parameter as a template.
+    if (diff < 0) {
+      // If there are N parameters but a single closure argument
```

Comment:
I would maybe factor out the body of this `if` into an addMissingArguments() function, then the comment is not even necessary

Reply:
Sure, I was planing to extract it as part of adding extraneous argument support but I can do it earlier too :)